androidtv/constants.py

```diff
@@ -28,20 +28,32 @@
 #: Determine whether the device is awake
 CMD_AWAKE = "dumpsys power | grep mWakefulness | grep -q Awake"
 
-#: Get the current app
-CMD_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep mCurrentFocus) && CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP"
+#: Parse current application identifier from dumpsys output and assign it to ``CURRENT_APP`` variable (assumes dumpsys output is momentarily set to ``CURRENT_APP`` variable)
+CMD_PARSE_CURRENT_APP = 'CURRENT_APP=${CURRENT_APP#*ActivityRecord{* * } CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && CURRENT_APP=${CURRENT_APP%\\}*}'
 
-#: Get the current app for a Google TV device
-CMD_CURRENT_APP_GOOGLE_TV = "CURRENT_APP=$(dumpsys activity a . | grep -E 'mResumedActivity' | cut -d ' ' -f 8) && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP"
+#: Assign focused application identifier to ``CURRENT_APP`` variable
+CMD_DEFINE_CURRENT_APP_VARIABLE = "CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && " + CMD_PARSE_CURRENT_APP
+
+#: Output identifier for current/focused application
+CMD_CURRENT_APP = CMD_DEFINE_CURRENT_APP_VARIABLE + ' && echo $CURRENT_APP'
+
+#: Assign focused application identifier to ``CURRENT_APP`` variable (for a Google TV device)
+CMD_DEFINE_CURRENT_APP_VARIABLE_GOOGLE_TV = 'CURRENT_APP=$(dumpsys activity a . | grep mResumedActivity) && ' + CMD_PARSE_CURRENT_APP
+
+#: Output identifier for current/focused application (for a Google TV device)
+CMD_CURRENT_APP_GOOGLE_TV = CMD_DEFINE_CURRENT_APP_VARIABLE_GOOGLE_TV + ' && echo $CURRENT_APP'
 
 #: Get the HDMI input
-CMD_HDMI_INPUT = "dumpsys activity starter | grep -o 'HDMIInputService\\/HW[0-9]' -m 1 | grep -o 'HW[0-9]'"
+CMD_HDMI_INPUT = "dumpsys activity starter | grep -E -o '(ExternalTv|HDMI)InputService/HW[0-9]' | grep -o 'HW[0-9]'"
```

Comment:
1. What effect does removing the `\\` have? (I'm not great with regular expressions...)
2. Why remove the `-m 1`? If there are multiple matches, that would cause a problem in `get_properties()`.

Reply:
> * What effect does removing the `\\` have? (I'm not great with regular expressions...)

What was happening here before was an (unnecessary) escape of the `/` character in the grep pattern.
After removing Python escape (`\\/`) it would be `\/`, which is equivalent to `/` (seeing the pattern is not enclosed between slashes).

> * Why remove the `-m 1`? If there are multiple matches, that would cause a problem in `get_properties()`.

From what I could deduce there is only a single entry in `dumpsys activity starter`, usually assigned to `mIntent`, based from the [output from my Sony TV](https://github.com/JeffLIrion/python-androidtv/blob/3a60f3154a8d39f7659e1d7d8609512505d1e386/tests/test_constants.py#L142-L240) and the Philips 55POS9002 output posted by reporter in https://github.com/JeffLIrion/python-androidtv/issues/202.

Maybe you can add a test case for your device(s) or for additional comparison post here the output of `dumpsys activity starter` for your device(s) with having an HDMI input active?

Reply:
Thanks for the explanations. I think I'd prefer to keep the `-m 1`, just to be safe. I don't see how it could hurt anything.